### ISSUE TITLE
NavigationView night mode deletes SpeechAnnouncement listener

### ISSUE THREAD

**Issue by kiskunk** [mapbox/mapbox-navigation-android#2046]
I am using navigationView in a fragment, from the example. I added custom listeners to the NavigationViewOptions.Builder. Since navigationView cannot support a simple day/night mode change, the tutorials showed me, that I have to recreate the activity to set the AppCompatDelegate for the night mode. After recreating the activity, I start a new navigationView (since the previous one called onDestroy), and adding the custom SpeechAnnouncementListener, but this time, the navigationView doesn't care about it, it generates its own. It makes no difference usig "this", to implement the lsiteners, or writing the "new XYListener". Neither listeners gets called. If I put a breakpoint in them, after recreating the activity and the navigationView, the breakpoint will never be active.

```
NavigationViewOptions.Builder navigationOptionsBuilder = NavigationViewOptions.builder()
                    .navigationListener(this)
                    .directionsRoute(selectedRoute)
                    .shouldSimulateRoute(isSimulate)
                    /*.routeListener(this)*/
                    .progressChangeListener(this)
                    .instructionListListener(this)
                    .speechAnnouncementListener(new SpeechAnnouncementListener() {
                        @Override
                        public SpeechAnnouncement willVoice(SpeechAnnouncement announcement) {
                            if (PreferencesManager.getInstance().getLanguage().equals("hu")) {
                                saySomething(TranslatorUtil.getTranslationWithSeparator(announcement.announcement().toLowerCase()));
                                return null;
                            } else {
                                return announcement;
                            }
                        }
                    })
                    .bannerInstructionsListener(this)
                    .milestoneEventListener(this);

            navigationView.startNavigation(navigationOptionsBuilder.build());
```

**Android API:** 16-28
**Mapbox Navigation SDK version:** 0.40.0

### Steps to trigger behavior

 1. create navigationView in fragment with custom SpeechAnnouncementListener
 2.  put a breakpoint in it, and confirm, that the process stops when it reaches the listener's function
 3. recreate the activity
 4. confirm, that this time, the process will never stop at the breakpoint

### Expected behavior

After recreating the activity, and the whole fragment, the navigationView should use the given listener.

### Actual behavior

After recreating the activity, the navigationView never uses the given listener.

**Comment by Guardiola31337** [comment]
@abhishek1508 @JunDai is this still an issue in `1.0`? Could you confirm?

**Comment by abhishek1508** [comment]
Let's verify this.
cc @JunDai @cafesilencio @langsmith

**Comment by cafesilencio** [comment]
I've tested with our soon to be released 1.0 SDK. Using one of our examples I added a speech announcement listener to the navigation view running in a fragment. I've tested putting the app in the background and bringing back to the foreground as well as rotating the device and I can see my logging message in the lister I added.  Thus far I'm not able to reproduce the issue described. 

Please be on the lookout for our first public preview of UI SDK v1.0 coming soon. Feel free to create a new issue if you feel it still exists.
